rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -232,8 +81,73 @@ void mic_context::operator()()
 	}
 }
 
-s32 microphone_device::open_microphone()
+void mic_context::load_config_and_init()
+{
+	auto device_list = fmt::split(g_cfg.audio.microphone_devices, {"@@@"});
+
+	if (device_list.size())
+	{
+		switch (g_cfg.audio.microphone_type)
+		{
+		case microphone_handler::standard:
+		{
+			for (u32 index = 0; index < device_list.size(); index++)
+			{
+				mic_list.emplace(std::piecewise_construct, std::forward_as_tuple(index), std::forward_as_tuple(microphone_handler::standard));
+				mic_list.at(index).add_device(device_list[index]);
+			}
+			break;
+		}
+		case microphone_handler::singstar:
+		{
+			mic_list.emplace(std::piecewise_construct, std::forward_as_tuple(0), std::forward_as_tuple(microphone_handler::singstar));
+			mic_list.at(0).add_device(device_list[0]);
+			if (device_list.size() >= 2)
+				mic_list.at(0).add_device(device_list[1]);
+			break;
+		}
+		case microphone_handler::real_singstar:
+		case microphone_handler::rocksmith:
+		{
+			mic_list.emplace(std::piecewise_construct, std::forward_as_tuple(0), std::forward_as_tuple(g_cfg.audio.microphone_type));
+			mic_list.at(0).add_device(device_list[0]);
+			break;
+		}
+		default: break;
+		}
+	}
+}
+
+// Static functions
+
+void microphone_device::variable_byteswap(const void* src, void* dst, const u32 bytesize)
+{
+	switch (bytesize)
+	{
+	case 4: *(u32*)dst = se_storage<u32>::swap(*((be_t<u32>*)src)); break;
```

Comment:
It is more correct to use be_t alone (no manual swap), because it obeys both IS_LE_MACHINE and IS_BE_MACHINE.

Reply:
also bugged.

Reply:
Fixed thanks!